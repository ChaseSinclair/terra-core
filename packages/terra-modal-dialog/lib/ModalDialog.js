'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _terraModal = require('terra-modal');

var _terraModal2 = _interopRequireDefault(_terraModal);

var _ModalDialogHeader = require('./ModalDialogHeader');

var _ModalDialogHeader2 = _interopRequireDefault(_ModalDialogHeader);

var _ModalDialogContentHeader = require('./ModalDialogContentHeader');

var _ModalDialogContentHeader2 = _interopRequireDefault(_ModalDialogContentHeader);

var _ModalDialogContent = require('./ModalDialogContent');

var _ModalDialogContent2 = _interopRequireDefault(_ModalDialogContent);

var _ModalDialogFooter = require('./ModalDialogFooter');

var _ModalDialogFooter2 = _interopRequireDefault(_ModalDialogFooter);

require('./ModalDialog.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  /**
   * String that labels the modal for screen readers
   **/
  ariaLabel: _propTypes2.default.string.isRequired,
  /**
   * Content inside the modal dialog
   **/
  children: _propTypes2.default.node.isRequired,
  /**
   * CSS classnames that are append to the modal
   **/
  classNameModal: _propTypes2.default.string,
  /**
   * CSS classnames that are append to the overlay
   **/
  classNameOverlay: _propTypes2.default.string,
  /**
   * If set to true, the modal will close when the esc key is pressed
   **/
  closeOnEsc: _propTypes2.default.bool,
  /**
   * If set to true, the modal will close when a mouseclick is triggered outside the modal
   **/
  closeOnOutsideClick: _propTypes2.default.bool,
  /**
   * If set to true, the modal will be fullscreen on all breakpoint sizes
   **/
  isFullscreen: _propTypes2.default.bool,
  /**
   * If set to true, the modal will rendered as opened
   **/
  isOpen: _propTypes2.default.bool.isRequired,
  /**
   * Function to set isOpen={false} and close the modal.
   **/
  onRequestClose: _propTypes2.default.func.isRequired,
  /**
   * Role attribute on the modal dialog
   **/
  role: _propTypes2.default.string
};

var defaultProps = {
  ariaLabel: null,
  children: null,
  classNameModal: null,
  classNameOverlay: null,
  closeOnEsc: true,
  closeOnOutsideClick: true,
  isFullscreen: false,
  isOpen: false,
  role: 'document'
};

var ModalDialog = function (_React$Component) {
  _inherits(ModalDialog, _React$Component);

  function ModalDialog() {
    _classCallCheck(this, ModalDialog);

    var _this = _possibleConstructorReturn(this, (ModalDialog.__proto__ || Object.getPrototypeOf(ModalDialog)).call(this));

    _this.state = { height: '30px' };
    _this.calculateHeight = _this.calculateHeight.bind(_this);
    return _this;
  }

  _createClass(ModalDialog, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.calculateHeight();
      window.addEventListener('resize', this.calculateHeight);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      window.removeEventListener('resize', this.calculateHeight);
    }
  }, {
    key: 'calculateHeight',
    value: function calculateHeight() {
      // When the width <=767, the modal changes to fullscreen
      if (document.documentElement.clientWidth > 767) {
        this.setState({ height: document.documentElement.clientHeight - 100 });
      } else {
        this.setState({ height: null });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          ariaLabel = _props.ariaLabel,
          children = _props.children,
          classNameModal = _props.classNameModal,
          classNameOverlay = _props.classNameOverlay,
          closeOnEsc = _props.closeOnEsc,
          closeOnOutsideClick = _props.closeOnOutsideClick,
          isFullscreen = _props.isFullscreen,
          isOpen = _props.isOpen,
          role = _props.role,
          onRequestClose = _props.onRequestClose,
          customProps = _objectWithoutProperties(_props, ['ariaLabel', 'children', 'classNameModal', 'classNameOverlay', 'closeOnEsc', 'closeOnOutsideClick', 'isFullscreen', 'isOpen', 'role', 'onRequestClose']);

      var newHeight = { maxHeight: this.state.height };

      if (!isOpen) {
        return null;
      }

      return _react2.default.createElement(
        _terraModal2.default,
        {
          ariaLabel: ariaLabel,
          classNameModal: classNameModal,
          classNameOverlay: classNameOverlay,
          closeOnOutsideClick: closeOnOutsideClick,
          closeOnEsc: closeOnEsc,
          isFullscreen: isFullscreen,
          isOpen: isOpen,
          onRequestClose: onRequestClose,
          role: role
        },
        _react2.default.createElement(
          'div',
          _extends({ style: newHeight, className: 'terra-ModalDialog' }, customProps),
          children
        )
      );
    }
  }]);

  return ModalDialog;
}(_react2.default.Component);

ModalDialog.propTypes = propTypes;
ModalDialog.defaultProps = defaultProps;

ModalDialog.Header = _ModalDialogHeader2.default;
ModalDialog.ContentHeader = _ModalDialogContentHeader2.default;
ModalDialog.Content = _ModalDialogContent2.default;
ModalDialog.Footer = _ModalDialogFooter2.default;

exports.default = ModalDialog;