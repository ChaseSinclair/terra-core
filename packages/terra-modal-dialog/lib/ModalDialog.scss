@import '~terra-mixins';

.terra-ModalDialog {
  display: flex;
  flex-direction: column;
  height: 100%;
}
