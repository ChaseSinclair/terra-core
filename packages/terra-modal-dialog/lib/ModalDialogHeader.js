'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ModalDialogHeaderDismiss = require('./ModalDialogHeaderDismiss');

var _ModalDialogHeaderDismiss2 = _interopRequireDefault(_ModalDialogHeaderDismiss);

require('./ModalDialogHeader.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  children: _propTypes2.default.node,
  hasDismiss: _propTypes2.default.node,
  onClick: _propTypes2.default.func
};

var defaultProps = {
  children: null
};

var ModalDialogHeader = function ModalDialogHeader(_ref) {
  var children = _ref.children,
      onClick = _ref.onClick,
      customProps = _objectWithoutProperties(_ref, ['children', 'onClick']);

  var dismissButton = onClick ? _react2.default.createElement(_ModalDialogHeaderDismiss2.default, { onClick: onClick }) : null;

  return _react2.default.createElement(
    'div',
    { className: 'terra-ModalDialog-header' },
    _react2.default.createElement(
      'div',
      _extends({ className: 'terra-ModalDialog-header-title' }, customProps),
      children
    ),
    dismissButton
  );
};

ModalDialogHeader.propTypes = propTypes;
ModalDialogHeader.defaultProps = defaultProps;

exports.default = ModalDialogHeader;