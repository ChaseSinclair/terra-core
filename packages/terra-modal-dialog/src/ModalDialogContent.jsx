import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogContent.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogContent = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-content'} {...customProps} >
    {children}
  </div>
  );


ModalDialogContent.propTypes = propTypes;
ModalDialogContent.defaultProps = defaultProps;

export default ModalDialogContent;
