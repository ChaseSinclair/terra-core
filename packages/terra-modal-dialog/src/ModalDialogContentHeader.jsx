import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogContentHeader.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogContentHeader = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-contentHeader'} {...customProps} >
    {children}
  </div>
  );

ModalDialogContentHeader.propTypes = propTypes;
ModalDialogContentHeader.defaultProps = defaultProps;

export default ModalDialogContentHeader;
