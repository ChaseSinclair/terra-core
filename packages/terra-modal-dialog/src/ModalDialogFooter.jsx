import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogFooter.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogFooter = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-footer'} {...customProps} >
    {children}
  </div>
  );


ModalDialogFooter.propTypes = propTypes;
ModalDialogFooter.defaultProps = defaultProps;

export default ModalDialogFooter;

