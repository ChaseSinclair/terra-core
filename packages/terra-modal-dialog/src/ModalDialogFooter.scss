@import '~terra-mixins';

.terra-ModalDialog-footer {
  background-color: $terra-grey-5;
  border-top: 1px solid $terra-grey-15;
}

