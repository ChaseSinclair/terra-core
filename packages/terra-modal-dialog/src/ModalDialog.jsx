import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'terra-modal';
import Header from './ModalDialogHeader';
import ContentHeader from './ModalDialogContentHeader';
import Content from './ModalDialogContent';
import Footer from './ModalDialogFooter';
import './ModalDialog.scss';

const propTypes = {
  /**
   * String that labels the modal for screen readers
   **/
  ariaLabel: PropTypes.string.isRequired,
  /**
   * Content inside the modal dialog
   **/
  children: PropTypes.node.isRequired,
  /**
   * CSS classnames that are append to the modal
   **/
  classNameModal: PropTypes.string,
  /**
   * CSS classnames that are append to the overlay
   **/
  classNameOverlay: PropTypes.string,
  /**
   * If set to true, the modal will close when the esc key is pressed
   **/
  closeOnEsc: PropTypes.bool,
  /**
   * If set to true, the modal will close when a mouseclick is triggered outside the modal
   **/
  closeOnOutsideClick: PropTypes.bool,
  /**
   * If set to true, the modal will be fullscreen on all breakpoint sizes
   **/
  isFullscreen: PropTypes.bool,
  /**
   * If set to true, the modal will rendered as opened
   **/
  isOpen: PropTypes.bool.isRequired,
  /**
   * Function to set isOpen={false} and close the modal.
   **/
  onRequestClose: PropTypes.func.isRequired,
  /**
   * Role attribute on the modal dialog
   **/
  role: PropTypes.string,
};

const defaultProps = {
  ariaLabel: null,
  children: null,
  classNameModal: null,
  classNameOverlay: null,
  closeOnEsc: true,
  closeOnOutsideClick: true,
  isFullscreen: false,
  isOpen: false,
  role: 'document',
};

class ModalDialog extends React.Component {
  constructor() {
    super();
    this.state = ({ height: '30px' });
    this.calculateHeight = this.calculateHeight.bind(this);
  }

  componentDidMount() {
    this.calculateHeight();
    window.addEventListener('resize', this.calculateHeight);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.calculateHeight);
  }

  calculateHeight() {
    // When the width <=767, the modal changes to fullscreen
    if (document.documentElement.clientWidth > 767) {
      this.setState({ height: document.documentElement.clientHeight - 100 });
    } else {
      this.setState({ height: null });
    }
  }

  render() {
    const {
          ariaLabel,
          children,
          classNameModal,
          classNameOverlay,
          closeOnEsc,
          closeOnOutsideClick,
          isFullscreen,
          isOpen,
          role,
          onRequestClose,
           ...customProps } = this.props;

    const newHeight = { maxHeight: this.state.height };

    if (!isOpen) {
      return null;
    }

    return <Modal
              ariaLabel={ariaLabel}
              classNameModal={classNameModal}
              classNameOverlay={classNameOverlay}
              closeOnOutsideClick={closeOnOutsideClick}
              closeOnEsc={closeOnEsc}
              isFullscreen={isFullscreen}
              isOpen={isOpen}
              onRequestClose={onRequestClose}
              role={role}
           >
             <div style={newHeight} className="terra-ModalDialog" {...customProps}>
               {children}
             </div>
           </Modal>
  }
}

ModalDialog.propTypes = propTypes;
ModalDialog.defaultProps = defaultProps;

ModalDialog.Header = Header;
ModalDialog.ContentHeader = ContentHeader;
ModalDialog.Content = Content;
ModalDialog.Footer = Footer;

export default ModalDialog;
