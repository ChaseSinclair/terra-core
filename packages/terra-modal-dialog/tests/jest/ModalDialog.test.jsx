import React from 'react';
import ModalDialogExample from './ModalDialogExample';
import ModalDialogHeader from '../../src/ModalDialogHeader';
import ModalDialogContentHeader from '../../src/ModalDialogContentHeader';
import ModalDialogContent from '../../src/ModalDialogContent';
import ModalDialogFooter from '../../src/ModalDialogFooter';
import ModalDialog from '../../src/ModalDialog';

it('should shallow a modal dialog', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal).toMatchSnapshot();
});

it('should mount a modal dialog', () => {
  const modal = mount(<ModalDialogExample />);
  expect(modal).toMatchSnapshot();
});

it('should render a modal dialog', () => {
  const modal = render(<ModalDialogExample />);
  expect(modal).toMatchSnapshot();
});

it('should have a header', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialogHeader).exists()).toEqual(true);
});

it('should have a content header', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialogContentHeader).exists()).toEqual(true);
});

it('should have content', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialogContent).exists()).toEqual(true);
});

it('should have a footer', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialogFooter).exists()).toEqual(true);
});

it('should have a ariaLabel prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('ariaLabel')).toEqual("Modal Dialog Example");
});

it('should have a classNameModal prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('classNameModal')).toEqual(null);
});

it('should have a classNameOverlay prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('classNameOverlay')).toEqual(null);
});

it('should have a closeOnEsc prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('closeOnEsc')).toEqual(true);
});

it('should have a closeOnOutsideClick prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('closeOnOutsideClick')).toEqual(true);
});

it('should have an isFullscreen prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('isFullscreen')).toEqual(false);
});

it('should have an isOpen prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('isOpen')).toEqual(true);
});

it('should have a role prop', () => {
  const modal = shallow(<ModalDialogExample />);
  expect(modal.find(ModalDialog).prop('role')).toEqual('document');
});

