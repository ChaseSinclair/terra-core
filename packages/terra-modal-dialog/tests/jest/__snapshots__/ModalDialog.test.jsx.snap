exports[`test should mount a modal dialog 1`] = `
<ModalDialogExample>
  <div>
    <ModalDialog
      ariaLabel="Modal Dialog Example"
      classNameModal={null}
      classNameOverlay={null}
      closeOnEsc={true}
      closeOnOutsideClick={true}
      isFullscreen={false}
      isOpen={true}
      onRequestClose={[Function]}
      role="document">
      <Modal
        ariaLabel="Modal Dialog Example"
        classNameModal={null}
        classNameOverlay={null}
        closeOnEsc={true}
        closeOnOutsideClick={true}
        isFullscreen={false}
        isOpen={true}
        isScrollable={false}
        onRequestClose={[Function]}
        role="document">
        <Portal
          isOpened={true}
          onClose={[Function]}
          onOpen={[Function]}
          onUpdate={[Function]} />
      </Modal>
    </ModalDialog>
    <button
      onClick={[Function]}>
      Open Modal
    </button>
  </div>
</ModalDialogExample>
`;

exports[`test should render a modal dialog 1`] = `
<div>
  <button>
    Open Modal
  </button>
</div>
`;

exports[`test should shallow a modal dialog 1`] = `
<div>
  <ModalDialog
    ariaLabel="Modal Dialog Example"
    classNameModal={null}
    classNameOverlay={null}
    closeOnEsc={true}
    closeOnOutsideClick={true}
    isFullscreen={false}
    isOpen={true}
    onRequestClose={[Function]}
    role="document">
    <ModalDialogHeader
      onClick={[Function]}>
      Modal Dialog Header
    </ModalDialogHeader>
    <ModalDialogContentHeader>
      Static Content Header
    </ModalDialogContentHeader>
    <ModalDialogContent>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut finibus velit non vehicula gravida. Nullam tempor erat at eros interdum, eget sodales mi finibus. Donec ac lorem enim. Donec imperdiet arcu sit amet interdum egestas. Aliquam quam quam, accumsan et egestas sed, imperdiet et purus. Vestibulum lacus ante, pulvinar sit amet turpis non, ultrices molestie nunc. Nam mollis pharetra felis sit amet eleifend. Nunc sagittis aliquet efficitur. Integer gravida, augue in mollis ultrices, felis diam maximus turpis, vitae pretium augue libero sit amet leo. Curabitur vel magna ac nulla laoreet viverra ac et mauris. In eleifend ex quis ipsum tristique, ut efficitur ligula luctus. Nullam id pretium nisl. Sed nec vehicula dolor, quis facilisis odio. Sed volutpat magna ut tincidunt fermentum. Mauris cursus metus id laoreet consectetur.
      </p>
      <p>
        Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Nunc nisl ipsum, laoreet nec tincidunt ut, sodales eget tortor. Pellentesque pharetra mauris eget dignissim lobortis. Mauris nec convallis nunc. Morbi in accumsan lectus, a ultrices elit. Donec pharetra tristique tortor, in consectetur massa commodo auctor. Maecenas at sollicitudin odio, in vehicula arcu. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      </p>
      <p>
        Aliquam mattis urna et nulla suscipit, id consequat dolor gravida. Cras varius purus eu massa aliquet, nec suscipit erat volutpat. In sit amet cursus tellus. Sed sit amet massa id metus sagittis ornare at eu ligula. Donec nec mi vel ipsum vulputate porta. Integer et justo urna. Donec scelerisque hendrerit felis at finibus. Duis in est ac nulla sodales luctus non vitae lectus. Suspendisse vel libero vel nisl accumsan dapibus ac non ipsum.
      </p>
    </ModalDialogContent>
    <ModalDialogFooter>
      Footer
    </ModalDialogFooter>
  </ModalDialog>
  <button
    onClick={[Function]}>
    Open Modal
  </button>
</div>
`;
